### PR TITLE
Add Hippa Hitrust v9.2 compliance queries. Closes #52

### LINKED ISSUE #52: Add Hippa Hitrust v9.2 compliance queries
**Is your feature request related to a problem? Please describe.**
Add Hippa Hitrust v9.2 compliance queries 

**Describe the solution you'd like**
Queries to achieve the Hippa Hitrust compliance to be added.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by krishna5891**
### Checklist
- [x] Issue(s) linked

### CODE REVIEW

File: query/postgres/postgres_db_server_geo_redundat_backup_enabled.sql [hunk 1]
```diff
@@ -0,0 +1,19 @@
+select
+  -- Required Columns
+  s.id as resource,
+  case
+    when geo_redundant_backup = 'Enabled' then 'ok'
+    else 'alarm'
+  end as status,
+  case
+    when geo_redundant_backup = 'Enabled' then name || ' Geo redundat backup enabled.'
+    else  name || ' Geo redundat backup disabled.'
```
Comment (cbruno10) [inline query/postgres/postgres_db_server_geo_redundat_backup_enabled.sql:10]:
```suggestion
    else  name || ' Geo-redundant backup disabled.'
```

File: query/postgres/postgres_db_server_geo_redundat_backup_enabled.sql [hunk 2]
```diff
@@ -0,0 +1,19 @@
+select
```
Comment (cbruno10) [inline query/postgres/postgres_db_server_geo_redundat_backup_enabled.sql:1]: Please rename file to `query/postgres/postgres_db_server_geo_redundant_backup_enabled.sql`

File: query/postgres/postgres_db_server_geo_redundat_backup_enabled.sql [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  -- Required Columns
+  s.id as resource,
+  case
+    when geo_redundant_backup = 'Enabled' then 'ok'
+    else 'alarm'
+  end as status,
+  case
+    when geo_redundant_backup = 'Enabled' then name || ' Geo redundat backup enabled.'
```
[the remark this answers is not in the document]
Comment (krishna5891) [reply query/postgres/postgres_db_server_geo_redundat_backup_enabled.sql]: Updated as suggested.

File: query/postgres/postgres_db_server_geo_redundat_backup_enabled.sql (hunk 1 above)
Comment (krishna5891) [reply query/postgres/postgres_db_server_geo_redundat_backup_enabled.sql]: Updated as suggested.

File: query/postgres/postgres_db_server_geo_redundat_backup_enabled.sql (hunk 2 above)
Comment (krishna5891) [reply query/postgres/postgres_db_server_geo_redundat_backup_enabled.sql]: Updated as suggested.

File: query/network/network_security_group_remote_access_restricted.sql [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    sg -> 'properties' ->> 'access' = 'Allow'
+    and sg -> 'properties' ->> 'direction' = 'Inbound'
+    and sg -> 'properties' ->> 'protocol' = 'TCP'
+    and sip in ('*', '0.0.0.0', '0.0.0.0/0', 'Internet', 'any', '<nw>/0', '/0')
+    and (
+      dport in ('22', '3389', '*')
+      or (
+        dport like '%-%'
+        and split_part(dport, '-', 1) :: integer <= 3389
+        and split_part(dport, '-', 2) :: integer >= 3389
```
Comment (khushboo9024) [inline query/network/network_security_group_remote_access_restricted.sql:19]: why we are not handling this part for port `22` ?

File: query/iam/iam_subscription_owner_max_3.sql [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+)
+select
+  -- Required columns
+  owner.subscription_id as resource,
+  case
+    when count(*) >= 3 then 'ok'
+    else 'alarm'
+  end as status,
+  case
+    when count(*) = 1 then 'There is 1 owner.'
```
Comment (rajlearner17) [inline query/iam/iam_subscription_owner_max_3.sql:20]: There is 1 owner. > Only 1 owner associated.

File: query/iam/iam_subscription_owner_more_than_1.sql [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+)
+select
+  -- Required columns
+  owner.subscription_id as resource,
+  case
+    when count(*) > 1 then 'ok'
+    else 'alarm'
+  end as status,
+    case
+    when count(*) = 1 then 'There is 1 owner.'
```
Comment (rajlearner17) [inline query/iam/iam_subscription_owner_more_than_1.sql:20]: There is 1 owner. > Only 1 owner associated.

File: query/iam/iam_subscription_owner_max_3.sql [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  from
+    azure_role_definition
+  where
+    role_name = 'Owner'
+)
+select
+  -- Required columns
+  owner.subscription_id as resource,
+  case
+    when count(*) >= 3 then 'ok'
```
Comment (rajlearner17) [inline query/iam/iam_subscription_owner_max_3.sql:16]: It is recommended to designate up to 3 subscription owners in order to reduce the potential for breach by a compromised owner.

A maximum of 3 owners should be designated for your subscription

File: query/network/network_security_group_remote_access_restricted.sql (hunk 4 above)
Comment (krishna5891) [reply query/network/network_security_group_remote_access_restricted.sql]: port 22 added on this part as well.

File: query/iam/iam_subscription_owner_max_3.sql (hunk 5 above)
Comment (krishna5891) [reply query/iam/iam_subscription_owner_max_3.sql]: updated the reason as per the suggestion.

File: query/iam/iam_subscription_owner_more_than_1.sql (hunk 6 above)
Comment (krishna5891) [reply query/iam/iam_subscription_owner_more_than_1.sql]: updated the reason as per the suggestion.

File: query/iam/iam_subscription_owner_max_3.sql (hunk 7 above)
Comment (krishna5891) [reply query/iam/iam_subscription_owner_max_3.sql]: changed the query to handle maximum of 3 owners per subscription.

File: query/appservice/appservice_web_app_remote_debugging_disabled.sql [hunk 8]
```diff
@@ -0,0 +1,19 @@
+select
+  -- Required Columns
+  app.id as resource,
+  case
+    when configuration -> 'properties' ->> 'remoteDebuggingEnabled' = 'false' then 'alarm'
```
Comment (khushboo9024) [inline query/appservice/appservice_web_app_remote_debugging_disabled.sql:5]: I think we should give `alarm` when   'remoteDebuggingEnabled' = 'true'.
Please check once.
https://docs.microsoft.com/en-us/azure/governance/policy/samples/hipaa-hitrust-9-2#ports-services-and-similar-applications-installed-on-a-computer-or-network-systems-which-are-not-specifically-required-for-business-functionality-are-disabled-or-removed

Comment (krishna5891) [reply query/appservice/appservice_web_app_remote_debugging_disabled.sql]: made the changes according to the requirement

File: query/iam/iam_subscription_owner_more_than_1.sql [hunk 9]
```diff
@@ -0,0 +1,29 @@
+with owner_roles as (
+  select
+    role_name,
+    role_type, name,
+    title,
+    subscription_id
+  from
+    azure_role_definition
```
Comment (khushboo9024) [inline query/iam/iam_subscription_owner_more_than_1.sql:8]: we need to check from `azure_role_assignment` whether `Owner` azure_role_definition is assigned to users.
Please check this once.

Comment (krishna5891) [reply query/iam/iam_subscription_owner_more_than_1.sql]: updated the script using the azure_role_definition and azure_role_assignment tables

File: query/network/network_security_group_remote_access_restricted.sql [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+select
+  -- Required Columns
+  sg.id resource,
+  case
+    when nsg.sg_name is null then 'ok'
+    else 'alarm'
+  end as status,
+  case
+    when nsg.sg_name is null then sg.title || ' restricts Remote access from internet.'
+    else sg.title || ' allows Remote access from internet.'
```
Comment (Subhajit97) [inline query/network/network_security_group_remote_access_restricted.sql:40]:
```suggestion
    when nsg.sg_name is null then sg.title || ' restricts remote access from internet.'
    else sg.title || ' allows remote access from internet.'
```

Comment (krishna5891) [reply query/network/network_security_group_remote_access_restricted.sql]: updated the corrections

File: query/iam/iam_subscription_owner_max_3.sql [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+    d.role_type,
+    d.name,
+    d.title,
+    d.subscription_id,
+    a.principal_id as principal_id
+  from
+    azure_role_definition as d
+    left join azure_role_assignment as a on d.id = a.role_definition_id
+  where
+    d.role_name = 'Owner'
```
Comment (Subhajit97) [inline query/iam/iam_subscription_owner_max_3.sql:13]: Are we also including `Service Principals` in this check?
As per the docs, we should exclude `Service Principals`.

File: query/postgres/postgres_db_server_geo_redundant_backup_enabled.sql [hunk 12]
```diff
@@ -0,0 +1,19 @@
+select
+  -- Required Columns
+  s.id as resource,
+  case
+    when geo_redundant_backup = 'Enabled' then 'ok'
+    else 'alarm'
+  end as status,
+  case
+    when geo_redundant_backup = 'Enabled' then name || ' Geo-redundant backup enabled.'
+    else  name || ' Geo-redundant backup disabled.'
```
Comment (Subhajit97) [inline query/postgres/postgres_db_server_geo_redundant_backup_enabled.sql:10]:
```suggestion
    else name || ' Geo-redundant backup disabled.'
```

File: query/iam/iam_subscription_owner_max_3.sql (hunk 11 above)
Comment (krishna5891) [reply query/iam/iam_subscription_owner_max_3.sql]: we are not using the principals as part of query. it was used only in the select part, hence removed it from the select part as well.

File: query/postgres/postgres_db_server_geo_redundant_backup_enabled.sql (hunk 12 above)
Comment (krishna5891) [reply query/postgres/postgres_db_server_geo_redundant_backup_enabled.sql]: updated as suggested.

**Review by rajlearner17 — APPROVED** [review]
LGTM to merge and revalidate

**Pull request merged**